modules/openapi-generator/src/main/resources/aspnetcore/2.0/controller.mustache

```diff
@@ -23,13 +23,15 @@ namespace {{packageName}}.Controllers
         /// <param name="{{paramName}}">{{description}}</param>{{/allParams}}{{#responses}}
         /// <response code="{{code}}">{{message}}</response>{{/responses}}
         [{{httpMethod}}]
-        [Route("{{{basePathWithoutHost}}}{{{path}}}")]
-        [ValidateModelState]{{#useSwashbuckle}}
+        [Route("{{{basePathWithoutHost}}}{{{path}}}")]{{#hasConsumes}}{{^consumes.0.x-mediaIsWildcard}}
+        [Consumes({{#consumes}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/consumes}})]{{/consumes.0.x-mediaIsWildcard}}{{/hasConsumes}}{{#hasProduces}}{{^produces.0.x-mediaIsWildcard}}
+        [Produces({{#produces}}"{{mediaType}}"{{#hasMore}}, {{/hasMore}}{{/produces}})]
+        {{/produces.0.x-mediaIsWildcard}}{{/hasProduces}}[ValidateModelState]{{#useSwashbuckle}}
         [SwaggerOperation("{{operationId}}")]{{#responses}}{{#dataType}}
         [SwaggerResponse(statusCode: {{code}}, type: typeof({{&dataType}}), description: "{{message}}")]{{/dataType}}{{^dataType}}{{/dataType}}{{/responses}}{{/useSwashbuckle}}
         public virtual IActionResult {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
-        { {{#responses}}
-{{#dataType}}
+        {
+{{^usePartialControllerClass}}{{#responses}}{{#dataType}}
```

Comment:
You may want do the following instead:
```
{{^usePartialControllerClass}}
  {{#responses}}
  {{#dataType}}
```
and in line 41
```
  {{/dataType}}
  {{/responses}}
```
so that the auto-generated code will look better without leading spaces, etc